fabric-resource-loader-v1/src/main/java/net/fabricmc/fabric/impl/resource/v1/ResourceReloaderPhaseData.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.resource.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.resource.ResourceReloader;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.base.toposort.SortableNode;
+
+class ResourceReloaderPhaseData extends SortableNode<ResourceReloaderPhaseData> {
+	final Identifier id;
+	ResourceReloader resourceReloader;
+	VanillaStatus vanillaStatus = VanillaStatus.NONE;
+
+	ResourceReloaderPhaseData(Identifier id, @Nullable ResourceReloader resourceReloader) {
+		super();
+		this.id = id;
+		this.resourceReloader = resourceReloader;
+	}
+
+	/**
+	 * Marks this phase and all preceding phases as running before Vanilla.
+	 */
+	void markBefore() {
+		boolean isAfter = this.vanillaStatus == VanillaStatus.AFTER;
+
+		if (this.vanillaStatus != VanillaStatus.NONE && !isAfter) return;
```

Comment:
Just to make sure, if `isAfter` is true this is a cyclic dependency, right?

Reply:
Indeed, it would be a cyclic dependency.